uefi/flash.go

```diff
@@ -84,6 +109,88 @@ func (f *FlashImage) Validate() []error {
 	return errors
 }
 
+// Extract extracts the flash image to the directory passed in.
+func (f *FlashImage) Extract(dirPath string) error {
+	absDirPath, err := filepath.Abs(dirPath)
+	if err != nil {
+		return err
+	}
+	// Create the directory if it doesn't exist
+	err = os.MkdirAll(absDirPath, 0755)
```

Comment:
if I am not mistaken, the various `BIOSRegion`, `MERegion` etc implement the `Firmware` interface (`Summary()` and `Validate()`). In this case, what about adding an `Extract` method to the `Firmware` interface? It would be more consistent, and here you would have to call `if err := region.Extract(); err != nil { ... }` for each region. The `Extract` method would also have to check whether the region is not `nil`

Reply:
It's actually been added to the Firmware interface already, but I think they're all different still. I could move the nil check into the extract functions for each of the regions, but I'm not sure what that gains us.